src/adapters/api-blueprint-adapter.coffee

```diff
@@ -298,9 +297,9 @@ legacyResourcesFrom1AResource = (legacyUrlConverterFn, resource, sourcemap, opti
       legacyResource.model = {}
 
     legacyResource.resourceParameters = resourceParameters or []
-    legacyResource.actionParameters   = actionParameters or []
+    legacyResource.actionParameters   = getParametersOf(action, options) or []
 
-    legacyResource.parameters = actionParameters or resourceParameters or []
+    legacyResource.parameters = legacyResource.resourceParameters.concat(legacyResource.actionParameters)
```

Comment:
The resource parameters should be the resource parameters always.

If there is a parameters section in the action it should be used for the action. Otherwise we should use the parameters section from the resource in the action.